packages/remix/src/utils/utils.ts

```diff
@@ -24,8 +24,15 @@ export function assertValidClerkState(val: any): asserts val is ClerkState | und
   }
 }
 
-type CloudflareEnv = {
-  env: Record<string, string>;
+type CloudflareEnv = { env: Record<string, string> };
+
+// https://remix.run/blog/remix-vite-stable#cloudflare-pages-support
+const hasCloudflareProxyContext = (context: any): context is { cloudflare: CloudflareEnv } => {
+  return !!context?.cloudflare?.env;
+};
+
+const hasCloudflareContext = (context: any): context is CloudflareEnv => {
+  return !!context?.env;
```

Comment:
❓ I think this check is not CloudflareContext specific. Isn't the `context.env` populated in other cases too?

Reply:
Afaik, only CF prepopulates .env